core/src/adjustCaretPosition.js

```diff
@@ -50,7 +50,7 @@ export default function adjustCaretPosition({
   // positioning.
   const possiblyHasRejectedChar = isAddition && (
     previousConformedValue === conformedValue ||
-    conformedValue === placeholder
+    (conformedValue === placeholder && !placeholder.includes('0'))
```

Comment:
@ethankong113 

` && !placeholder.includes('0'))` 

what is this for?  It's not obvious why its there, and and it doesn't seem to make any difference to your issue (that I can tell).. and there are no tests that cover it.  Is it needed?

Reply:
the conformedValue after transforming ['0', decimalSymbol, caretTrap] would be the same as the placeholder.. React Mask would not detect any difference and you cannot input anything if you type "."